ZuluContent/Zulu/Engines/Loot/LootTable.cs

```diff
@@ -77,9 +79,18 @@ public List<LootItem> Roll()
             {
                 li.ItemLevel = ItemLevel;
                 li.ItemChance = ItemChance;
+                LootItem dictLi;
+                if (lootItemsDict.TryGetValue(li.Type.ToString(), out dictLi))
```

Comment:
Had some issues here with the reagents not stacking so this was my attempt at stacking them incase you're wondering

Reply:
Yeah this is a bit funky, we might be able to uncover a better approach here!

Firstly as a tip you can inline the out variable declaration to save space, e.g.

```csharp
LootItem dictLi;
if (lootItemsDict.TryGetValue(li.Type.ToString(), out dictLi))
```
Becomes
```csharp
if (lootItemsDict.TryGetValue(li.Type.ToString(), out var dictLi))
```

You also don't need to be converting the `LootItem.Type` property to string, you can just make the key type parameter `Type` in the dictionary for that. It shouldn't be any heavier perf wise and keeps you from passing strings around. e.g. 

```csharp
var lootItemsDict = new Dictionary<Type, LootItem>();
// ...
lootItemsDict.Add(li.Type, li);
```

I think this might be clobbering the quantity of previously generated loot items. It may even possibly be trying to increase the quantity of non-stackables like armor/weapons etc. You might want to set a breakpoint and make sure it's not causing less items to be generated, e.g. if two katanas we're meant to be in the loot, but instead now one spawns with an ignored quantity of 2.
```csharp
if (lootItemsDict.TryGetValue(li.Type, out dictLi))
{
    dictLi.Quantity++;

   // Should probably be
    dictLi.Quantity += li.Quantity;
}
```

I think the issue here is that the way "Stackables" work is based on the [`Item.Stackable` boolean property](https://github.com/modernuo/ModernUO/blob/c58aad733d8cdbcf1d6d0a558bf13fcc7035b90b/Projects/Server/Items/Item.cs#L351) which is [only set in the constructor of the items](https://github.com/ZuluHotelAustralia/zuluhotel/blob/17ba414aeb451f1078ca58cd6c6db4c8d90edb21/ZuluContent/Items/Resources/Reagents/BaseReagent.cs#L11) AFAIK. So at this point in the loot generation we don't know what loot is stackable and what isn't. So this might be better solved at the point where the loot is generated and then added to the container where it can be merged/stacked.

Reply:
Done